canvasobject.go

```diff
@@ -79,3 +79,9 @@ type Focusable interface {
 type Shortcutable interface {
 	TypedShortcut(Shortcut)
 }
+
+// Cursorable describes any CanvasObject that needs a cursor change
+type Cursorable interface {
+	Cursor() Cursor
+	SetCursor(Cursor)
```

Comment:
I'm with @andydotxyz here: don't add `SetCursor()`. I don't think there is a real use case. IMO the cursor depends on the object's purpose and the purpose of an object does not change without the object being changed.